Modernize cmake

mharshe opened the pull request:
Update min version of cmake to 3.12. Set minimum python version to 3.6
Simplify cmake linking since it is transitive

Update cmakelists for codegenerator tests

Fix yaml dependencies for zmq & dds builds

Isolate yaml dependencies

Bugfix high performance event loop test

pablogh-2000 reviewed (commented):
Squash commits and check that all builds correctly locally and also with @tomasz3dk scripts.

mharshe commented:
> Squash commits and check that all builds correctly locally and also with @tomasz3dk scripts.

Squashed commits and added @tomasz3dk as a reviewer so he can test.
